### PR TITLE
Magpie20 hadoopversion

### PR DESCRIPTION
**Pull request by bpanneton**
Tested with Hadoop 3.0.3 and it seems to work.

### CODE REVIEW

**Comment by bpanneton** [comment]
Fixed to correct base.

**Comment by chu11** [comment]
Hate to be a bother, but could you add Hadoop 3.0.1 and 3.0.2 into the testsuite too & documentation (README.md & doc/README).  It doesn't matter if you haven't tested it.

**Comment by chu11** [comment]
Wait a second, this worked for you with or without the "no-local-dir" patch?  B/c I never got that patch working for Hadopo 3.0.  Is that patch necessary going forward for you?

**Comment by bpanneton** [comment]
I believe I patched it without issue. I can add the 3.0.1 and 3.0.2 in.

**Comment by bpanneton** [comment]
Let me check again

**Comment by bpanneton** [comment]
The patch applies cleanly and works for me. I was able to run TeraSort. I also tried without the patch and it failed due to not finding the correct configurations. I will update it for the other versions of Hadoop (however I'm not actually going to test 3.0.1 or 3.0.2)

**Comment by chu11** [comment]
ok, b/c the patch actually didn't work for me.  Could you try on hadoop 3.0.0?

**Comment by bpanneton** [comment]
Sure. Give me a few mins to check.

**Comment by bpanneton** [comment]
So it appears to work fine for me. In the magpie sbatch srun config make sure you set the following:

export MAGPIE_NO_LOCAL_DIR="yes"                                                                
export MAGPIE_LOCAL_DIR="${_LOCAL_PREFIX}/magpie"
export HADOOP_LOCAL_DIR="${_LOCAL_PREFIX}/hadoop"
export ZOOKEEPER_DATA_DIR="${_LOCAL_PREFIX}/zookeeper"
export ZOOKEEPER_LOCAL_DIR="${_LOCAL_PREFIX}/zookeeper"

**Comment by bpanneton** [comment]
Just pushed the rename of the patches as well.

File: README.md [hunk 1]
```diff
@@ -74,7 +74,8 @@ those versions.
 Hadoop - 2.2.0+, 2.3.0+, 2.4.0+, 2.4.1+, 2.5.0+, 2.5.1+, 2.5.2+,
          2.6.0+, 2.6.1+, 2.6.2+, 2.6.3+, 2.6.4+, 2.6.5+, 2.7.0+,
          2.7.1+, 2.7.2+, 2.7.3+, 2.7.4+, 2.7.5+, 2.8.0+, 2.8.1+,
-         2.8.2+, 2.8.3+, 2.8.4+, 2.9.0+, 2.9.1+
+         2.8.2+, 2.8.3+, 2.8.4+, 2.9.0+, 2.9.1+, 3.0.1+, 3.0.2+,
```
Comment (chu11) [inline README.md]: oops, i guess i forgot 3.0

File: testsuite/test-common.sh [hunk 2]
```diff
@@ -12,7 +12,7 @@ hadoop2Xjava17versions="2.6.0 2.6.1 2.6.2 2.6.3 2.6.4 2.6.5 2.7.0 2.7.1 2.7.2 2.
 hadoop2Xjava17versions_javaversion=${java17}
 hadoop2Xjava18versions="2.7.4 2.7.5"
 hadoop2Xjava18versions_javaversion=${java18}
-hadoop3Xjava17versions="3.0.0"
+hadoop3Xjava17versions="3.0.0 3.0.3"
```
Comment (chu11) [inline testsuite/test-common.sh]: 3.01 and 3.02 here too

File: testsuite/test-generate.sh [hunk 3]
```diff
@@ -104,6 +104,7 @@ hadoop_2_8_4=y
 hadoop_2_9_0=y
 hadoop_2_9_1=y
 hadoop_3_0_0=y
+hadoop_3_0_3=y
```
Comment (chu11) [inline testsuite/test-generate.sh]: and 3_0_1 and 3_0_2

**Comment by chu11** [comment]
ok will that's good that it works, I'm not sure why it didn't work for me so long ago.  I'll have to try again.  perhaps my testsuite didn't set something up properly.

**Comment by bpanneton** [comment]
Updated!

**Comment by chu11** [comment]
awesome, thanks!  BTW, are you just playing with this branch?  Or are you already beginning to use it?

**Comment by bpanneton** [comment]
We are trying to run with Python3 and ended up needing Spark 2.3.0 and Hive 3.0.0 was the only one working with it. Then Hive required Hadoop 3.0.0 as well. I got Spark and Hadoop working Zeppeling 0.8.0 using Python3. I'm working on integrating Adam's Hive changes into the newer one as well.

**Comment by chu11** [comment]
it ends up my no-local-dir unit tests have been broken for some time, b/c I normally don't run them .. doh

**Pull request merged**